### ISSUE TITLE
[Crash] VayraPersonBountyIntel:1070 causing a NPE crash

### ISSUE THREAD

**Issue by RkShaRkz** [RkShaRkz/Starsector-IntelliJ-Template#58]
### Mod Name:
Vayra Merged

### Mod Version:
3.3.12 I suppose

Rarasek's mod list:
```
{"enabledMods": [
  "pantera_ANewLevel40",
  "aip_cores",
  "all_the_domain_drones+NewDrones",
  "A_S-F",
  "ApproLight",
  "BVH",
  "BSC",
  "bc",
  "HMI_brighton",
  "ORK",
  "Csp",
  "Celestial Mount Circle",
  "combatactivators",
  "lw_console",
  "diableavionics",
  "Diktat Enhancement",
  "dex",
  "edshipyard",
  "EmergentThreats_Vice",
  "EmergentThreats_IX_Revival",
  "exoticatechnologies",
  "lyr_ehm",
  "XLU",
  "fleetsizebydp",
  "Goat_Aviation_Bureau",
  "GrandColonies",
  "GMDA",
  "HMI_SV",
  "HMI",
  "hte",
  "aerialcombatsuit",
  "Imperium",
  "timid_xiv",
  "JYD",
  "kyeltziv",
  "lost_sector",
  "LTA",
  "lw_lazylib",
  "lockedAndLoaded",
  "LLI",
  "lumos_machina",
  "lunalib",
  "exshippack",
  "MagicLib",
  "Mayasuran Navy",
  "mc_collection",
  "missinghegemonyauxiliaryships",
  "missingshipscivilian",
  "missingsuperships",
  "missingmidlineships",
  "missingpatherships",
  "missingshipspirate",
  "ms_proto",
  "MoreMilitaryMissions",
  "advent_seven",
  "ness_saw",
  "nexerelin",
  "OcuA",
  "oas",
  "pn",
  "PirateMiniMegaMod",
  "progressiveSMods",
  "assortment_of_things",
  "RetroLib",
  "roider",
  "sv",
  "salshiexp",
  "scalartech",
  "swp",
  "mayu_specialupgrades",
  "starwars_abeOfarabia",
  "scf",
  "tahlan",
  "Terraforming & Station Construction",
  "star_federation",
  "thong",
  "thongredacto",
  "TraverserDesignBureau",
  "underworld",
  "vic",
  "TouchOfVanilla_vri",
  "cabal-co",
  "x_rp_ai",
  "XhanEmpire",
  "YNA",
  "astroidships",
  "shaderLib"
]}
```

Rarasek's stack trace:
```
[Thread-3] ERROR com.fs.starfarer.combat.CombatMain - java.lang.NullPointerException
java.lang.NullPointerException
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1192)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1904)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.spawnFleet(VayraPersonBountyIntel.java:1070)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.<init>(VayraPersonBountyIntel.java:205)
	at data.scripts.campaign.intel.VayraPersonBountyManager.createEvent(VayraPersonBountyManager.java:331)
	at data.scripts.campaign.intel.VayraPersonBountyManager.advance(VayraPersonBountyManager.java:305)
	at data.scripts.VayraMergedModPlugin.onNewGameAfterTimePass(VayraMergedModPlugin.java:356)
	at com.fs.starfarer.campaign.save.CampaignGameManager.o00000(Unknown Source)
	at com.fs.starfarer.title.TitleScreenState.dialogDismissed(Unknown Source)
	at com.fs.starfarer.ui.0.dismiss(Unknown Source)
	at com.fs.starfarer.campaign.save.if.actionPerformed(Unknown Source)
	at com.fs.starfarer.ui.o000.buttonPressed(Unknown Source)
	at com.fs.starfarer.ui.int.O00000(Unknown Source)
	at com.fs.starfarer.ui.int.processInput(Unknown Source)
	at com.fs.starfarer.ui.W.o00000(Unknown Source)
	at com.fs.starfarer.BaseGameState.traverse(Unknown Source)
	at com.fs.state.AppDriver.begin(Unknown Source)
	at com.fs.starfarer.combat.CombatMain.main(Unknown Source)
	at com.fs.starfarer.StarfarerLauncher.super(Unknown Source)
	at com.fs.starfarer.sTarfarerLauncher$1.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by RkShaRkz** [comment]
v_e00's modlist:
```
{"enabledMods": [
  "PAGSM",
  "yunruindustries",
  "pantera_ANewLevel120R",
  "adjustable_skill_thresholds",
  "adversary",
  "all_the_domain_drones+NewDrones",
  "all_the_domain_drones+Vanilla",
  "A_S-F",
  "ARSWP",
  "Cryo_but_better",
  "aod_capital",
  "aod_vos",
  "aotd_vok",
  "apex_design",
  "apocrita_association",
  "ARC",
  "aria",
  "armaa",
  "zzarmaa_anime",
  "arthrships",
  "bashedfactions",
  "BVH",
  "BSC",
  "pigeonpun_bionicalteration",
  "bc",
  "blackrock_driveyards",
  "blobcatportrait",
  "yrutl",
  "bricky",
  "HMI_brighton",
  "ORK",
  "CaptainsLog",
  "CFT",
  "CJHM",
  "Csp",
  "chatterexp",
  "chatter",
  "cmutils",
  "lw_console",
  "customizablestarsystems",
  "dmodservices",
  "istl_dassaultmikoyan",
  "DemtoShippack",
  "bbplus_refits",
  "diablearmaaonics",
  "diableavionics",
  "Diktat Enhancement",
  "dex",
  "domhist",
  "edshipyard",
  "EmergentThreats_Vice",
  "EmergentThreats_IX_Revival",
  "seven_nexus",
  "exoticatechnologies",
  "lyr_ehm",
  "EXPLOSIVE_FUEL",
  "FSF_MilitaryCorporation",
  "FPE",
  "fob",
  "sun_fuel_siphoning",
  "fp",
  "GalacticConstellate",
  "FantasyM",
  "Goat_Aviation_Bureau",
  "GrandColonies",
  "HMI_SV",
  "HMI",
  "hte",
  "hpst",
  "hyperdiving",
  "sun_hyperdrive",
  "independent_interstellar",
  "aerialcombatsuit",
  "IndEvo",
  "Imperium",
  "timid_xiv",
  "JYD",
  "keruvim_shipyards",
  "knights_of_ludd",
  "kyeltziv",
  "lost_sector",
  "LTA",
  "lakhesismonarchy",
  "lw_lazylib",
  "ArkLeg",
  "leg",
  "logisticsNotifications",
  "luddenhance",
  "luddenhanceied",
  "lunalib",
  "exshippack",
  "Apex unofficial expansion",
  "athena_dreadnought",
  "athena_dreadnaughty_mk2",
  "athena_dreadnaughty",
  "Dabble mael",
  "Dassaylt Megallan unofficial expansion",
  "captinjoehenrymod",
  "FED unofficial expansion",
  "Imperium_unofficial_exp",
  "Neutrino_unofficial_exp",
"Pearson unofficial expansion",
  "Templars unofficial expansion",
  "VIC_unofficial_exp",
  "XhanEmpire unofficial expansion",
  "mag_protect",
  "mag_protect_soldier",
  "MagicLib",
  "Mayasuran Navy",
  "missinghegemonyauxiliaryships",
  "missingshipscivilian",
  "missingsuperships",
  "missingmidlineships",
  "missingpatherships",
  "missingshipspirate",
  "niko_moreCombatTerrainEffects",
  "more_hullmods",
  "MoreMilitaryMissions",
  "NSP",
  "advent_seven",
  "Neutrino",
  "nexerelin",
  "kayse_phaseships",
  "OcuA",
  "oas",
  "PirateMiniMegaMod",
  "particleengine",
  "pearson_exotronics",
  "wyv_planetaryShieldAccessControl",
  "pigeonpun_projectsolace",
  "QualityCaptains",
  "assortment_of_things",
  "SalvageEverything",
  "SBR",
  "SBR_arma",
  "scalartech",
  "Scrapyard",
  "secretsofthefrontier",
  "SEEKER",
  "bb_plus",
  "m_sephira_refits",
  "swp",
  "mayu_specialupgrades",
  "stardustventures",
  "sun_starship_legends",
  "niko_stationAugments",
  "stelnet",
  "hakkari3",
  "alcoholism",
  "tahlan",
  "Terraforming & Station Construction",
  "Templars",
  "star_federation",
  "the_vass",
  "TBJ",
  "thong",
  "thongredacto",
  "THI",
  "TraverserDesignBureau",
  "TTSC",
  "uaf_arma",
  "underworld",
  "uaf",
  "yunru_ussr",
  "US",
  "UGH",
  "vic",
  "vayramerged",
  "vdba",
  "vbc",
  "VDMEBC",
  "TouchOfVanilla_vri",
  "niko_morePlanetaryConditions",
  "which_industry",
  "whichmod",
  "XhanEmpire",
  "YNA",
  "yunru_arsenal",
  "yunru_cunt",
  "yunru_hegemon",
  "yunru_ht",
  "yunruhullmods",
  "yunru_midline_expansion",
  "yunru_ose",
  "yunru_pirate_collection",
  "yunruworlds",
  "yunrucore",
  "yrxp",
  "shaderLib",
  "tahlanadditions",
  "KFM3",
  "KGT1",
  "KGT4",
  "KGT6",
  "KGT2"
]}
```

v_e00's stacktrace:
```
1213217 [Thread-3] ERROR com.fs.starfarer.combat.CombatMain  - java.lang.NullPointerException
java.lang.NullPointerException
    at com.fs.starfarer.campaign.CharacterStats.setSkillLevel(Unknown Source)
    at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1183)
    at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1904)
    at data.scripts.campaign.intel.VayraPersonBountyIntel.spawnFleet(VayraPersonBountyIntel.java:1070)
    at data.scripts.campaign.intel.VayraPersonBountyIntel.<init>(VayraPersonBountyIntel.java:205)
    at data.scripts.campaign.intel.VayraPersonBountyManager.createEvent(VayraPersonBountyManager.java:331)
    at data.scripts.campaign.intel.VayraPersonBountyManager.advance(VayraPersonBountyManager.java:305)
    at data.scripts.VayraMergedModPlugin.onNewGameAfterTimePass(VayraMergedModPlugin.java:356)
    at com.fs.starfarer.campaign.save.CampaignGameManager.o00000(Unknown Source)
    at com.fs.starfarer.title.TitleScreenState.dialogDismissed(Unknown Source)
    at com.fs.starfarer.ui.O.dismiss(Unknown Source)
    at com.fs.starfarer.ui.impl.for.dismiss(Unknown Source)
    at com.fs.starfarer.campaign.save.if.actionPerformed(Unknown Source)
    at com.fs.starfarer.ui.o0O0.buttonPressed(Unknown Source)
    at com.fs.starfarer.ui.int.Ò00000(Unknown Source)
    at com.fs.starfarer.ui.int.processInput(Unknown Source)
    at com.fs.starfarer.ui.W.o00000(Unknown Source)
    at com.fs.starfarer.BaseGameState.traverse(Unknown Source)
    at com.fs.state.AppDriver.begin(Unknown Source)
    at com.fs.starfarer.combat.CombatMain.main(Unknown Source)
    at com.fs.starfarer.StarfarerLauncher.super(Unknown Source)
    at com.fs.starfarer.StarfarerLauncher$1.run(Unknown Source)
    at java.lang.Thread.run(Unknown Source)
```

**Comment by RkShaRkz** [comment]
Happened to me too just now on a brand new game.

Was trying to repro hullmod changes to ET, selected a 'grand fleet' with starting level 3, and bam.

```
454337 [Thread-3] ERROR com.fs.starfarer.combat.CombatMain  - java.lang.NullPointerException
java.lang.NullPointerException
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1192)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1904)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.spawnFleet(VayraPersonBountyIntel.java:1070)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.<init>(VayraPersonBountyIntel.java:205)
	at data.scripts.campaign.intel.VayraPersonBountyManager.createEvent(VayraPersonBountyManager.java:331)
	at data.scripts.campaign.intel.VayraPersonBountyManager.advance(VayraPersonBountyManager.java:305)
	at data.scripts.VayraMergedModPlugin.onNewGameAfterTimePass(VayraMergedModPlugin.java:356)
	at com.fs.starfarer.campaign.save.CampaignGameManager.o00000(Unknown Source)
	at com.fs.starfarer.title.TitleScreenState.dialogDismissed(Unknown Source)
	at com.fs.starfarer.ui.O.dismiss(Unknown Source)
```

**Comment by RkShaRkz** [comment]
And again.

Free start > Grand fleet > Commander (level 6)

```
454337 [Thread-3] ERROR com.fs.starfarer.combat.CombatMain  - java.lang.NullPointerException
java.lang.NullPointerException
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1192)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1904)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.spawnFleet(VayraPersonBountyIntel.java:1070)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.<init>(VayraPersonBountyIntel.java:205)
	at data.scripts.campaign.intel.VayraPersonBountyManager.createEvent(VayraPersonBountyManager.java:331)
	at data.scripts.campaign.intel.VayraPersonBountyManager.advance(VayraPersonBountyManager.java:305)
	at data.scripts.VayraMergedModPlugin.onNewGameAfterTimePass(VayraMergedModPlugin.java:356)
	at com.fs.starfarer.campaign.save.CampaignGameManager.o00000(Unknown Source)
	at com.fs.starfarer.title.TitleScreenState.dialogDismissed(Unknown Source)
	at com.fs.starfarer.ui.O.dismiss(Unknown Source)
	at com.fs.starfarer.ui.impl.for.dismiss(Unknown Source)
	at com.fs.starfarer.campaign.save.if.actionPerformed(Unknown Source)
	at com.fs.starfarer.ui.o0O0.buttonPressed(Unknown Source)
	at com.fs.starfarer.ui.int.Ò00000(Unknown Source)
	at com.fs.starfarer.ui.int.processInput(Unknown Source)
	at com.fs.starfarer.ui.o0O0.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
	at com.fs.starfarer.ui.o000.dispatchEventsToChildren(Unknown Source)
	at com.fs.starfarer.ui.o000.processInputImpl(Unknown Source)
	at com.fs.starfarer.coreui.k.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
	at com.fs.starfarer.ui.o000.dispatchEventsToChildren(Unknown Source)
	at com.fs.starfarer.ui.o000.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
	at com.fs.starfarer.E.I.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
	at com.fs.starfarer.ui.o000.dispatchEventsToChildren(Unknown Source)
	at com.fs.starfarer.ui.o000.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.O.processInputImpl(Unknown Source)
	at com.fs.starfarer.campaign.save.if.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
	at com.fs.starfarer.ui.o000.dispatchEventsToChildren(Unknown Source)
	at com.fs.starfarer.ui.o000.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
	at com.fs.starfarer.title.TitleScreenState.processInput(Unknown Source)
	at com.fs.starfarer.BaseGameState.traverse(Unknown Source)
	at com.fs.state.AppDriver.begin(Unknown Source)
	at com.fs.starfarer.combat.CombatMain.main(Unknown Source)
	at com.fs.starfarer.StarfarerLauncher.super(Unknown Source)
	at com.fs.starfarer.StarfarerLauncher$1.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by RkShaRkz** [comment]
And again. Repro steps confirmed.

Start new game > Free start > Grand fleet > Commander (level 6) > start game > select no skills > crash
```
515227 [Thread-3] ERROR com.fs.starfarer.combat.CombatMain  - java.lang.NullPointerException
java.lang.NullPointerException
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1192)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderAndOfficersV2(FleetFactoryV3.java:1016)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderAndOfficers(FleetFactoryV3.java:866)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.createFleet(FleetFactoryV3.java:423)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.spawnFleet(VayraPersonBountyIntel.java:1028)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.<init>(VayraPersonBountyIntel.java:205)
	at data.scripts.campaign.intel.VayraPersonBountyManager.createEvent(VayraPersonBountyManager.java:331)
	at data.scripts.campaign.intel.VayraPersonBountyManager.advance(VayraPersonBountyManager.java:305)
	at data.scripts.VayraMergedModPlugin.onNewGameAfterTimePass(VayraMergedModPlugin.java:356)
	at com.fs.starfarer.campaign.save.CampaignGameManager.o00000(Unknown Source)
	at com.fs.starfarer.title.TitleScreenState.dialogDismissed(Unknown Source)
	at com.fs.starfarer.ui.O.dismiss(Unknown Source)
	at com.fs.starfarer.ui.impl.for.dismiss(Unknown Source)
	at com.fs.starfarer.campaign.save.if.actionPerformed(Unknown Source)
	at com.fs.starfarer.ui.o0O0.buttonPressed(Unknown Source)
	at com.fs.starfarer.ui.int.Ò00000(Unknown Source)
	at com.fs.starfarer.ui.int.processInput(Unknown Source)
	at com.fs.starfarer.ui.o0O0.processInputImpl(Unknown Source)
	at com.fs.starfarer.ui.thissuper.processInput(Unknown Source)
```

**Comment by RkShaRkz** [comment]
And again, this time with just one ship and level 3 start.
```
1948809 [Thread-3] ERROR com.fs.starfarer.combat.CombatMain  - java.lang.NullPointerException
java.lang.NullPointerException
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.campaign.CharacterStats.refreshCharacterStatsEffects(Unknown Source)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderSkills(FleetFactoryV3.java:1192)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderAndOfficersV2(FleetFactoryV3.java:1016)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.addCommanderAndOfficers(FleetFactoryV3.java:866)
	at com.fs.starfarer.api.impl.campaign.fleets.FleetFactoryV3.createFleet(FleetFactoryV3.java:423)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.spawnFleet(VayraPersonBountyIntel.java:1033)
	at data.scripts.campaign.intel.VayraPersonBountyIntel.<init>(VayraPersonBountyIntel.java:205)
	at data.scripts.campaign.intel.VayraPersonBountyManager.createEvent(VayraPersonBountyManager.java:331)
	at data.scripts.campaign.intel.VayraPersonBountyManager.advance(VayraPersonBountyManager.java:305)
	at data.scripts.VayraMergedModPlugin.onNewGameAfterTimePass(VayraMergedModPlugin.java:366)
	at com.fs.starfarer.campaign.save.CampaignGameManager.o00000(Unknown Source)
	at com.fs.starfarer.title.TitleScreenState.dialogDismissed(Unknown Source)
	at com.fs.starfarer.ui.O.dismiss(Unknown Source)
	at com.fs.starfarer.ui.impl.for.dismiss(Unknown Source)
	at com.fs.starfarer.campaign.save.if.actionPerformed(Unknown Source)
```

**Comment by RkShaRkz** [comment]
Doesn't seem to happen when starting out with factions, seems to happen only with "no faction (hard)" start

**Comment by RkShaRkz** [comment]
Yep, I ~~fixed~~ plugged this so the game no longer crashes. hooray.

**Comment by RkShaRkz** [comment]
Resolved by #60 in commit a59d99f - closing.
